arroyo-connectors/src/lib.rs

```diff
@@ -120,6 +120,7 @@ pub trait Connector: Send {
         name: &str,
         options: &mut HashMap<String, String>,
         schema: Option<&ConnectionSchema>,
+        profile: Option<&ConnectionProfile>,
```

Comment:
Could this be `Self::ProfileT` instead of `ConnectionProfile`, like it is in `from_config`? That would remove the responsibility of the individual connectors for having to do the deserialization and handle errors.